test/cqlpy/cassandra_tests/validation/operations/cql_vector_test.py

```diff
@@ -0,0 +1,478 @@
+    
+
+# Test commented out because it uses internal Cassandra Java APIs, not CQL
+#     @Test
+#     public void functions()
+#     {
+#         VectorType<Integer> type = VectorType.getInstance(Int32Type.instance, 2);
+#         Vector<Integer> vector = vector(1, 2);
+
+#         NativeFunctions.instance.add(new NativeScalarFunction("f", type, type)
+#         {
+#             @Override
+#             public ByteBuffer execute(Arguments arguments) throws InvalidRequestException
+#             {
+#                 return arguments.get(0);
+#             }
+
+#             @Override
+#             public Arguments newArguments(ProtocolVersion version)
+#             {
+#                 return FunctionArguments.newNoopInstance(version, 1);
+#             }
+#         });
+
+#         createTable(KEYSPACE, "CREATE TABLE %s (pk int primary key, value vector<int, 2>)");
+#         execute("INSERT INTO %s (pk, value) VALUES (0, ?)", vector);
+
+#         assertRows(execute("SELECT f(value) FROM %s WHERE pk=0"), row(vector));
+#         assertRows(execute("SELECT f([1, 2]) FROM %s WHERE pk=0"), row(vector));
+#     }
+
+# Test commented out because it uses internal Cassandra Java APIs, not CQL
+#     @Test
+#     public void specializedFunctions()
+#     {
+#         VectorType<Float> type = VectorType.getInstance(FloatType.instance, 2);
+#         Vector<Float> vector = vector(1.0f, 2.0f);
+
+#         NativeFunctions.instance.add(new NativeScalarFunction("f", type, type, type)
+#         {
+#             @Override
+#             public ByteBuffer execute(Arguments arguments) throws InvalidRequestException
+#             {
+#                 float[] left = arguments.get(0);
+#                 float[] right = arguments.get(1);
+#                 int size = Math.min(left.length, right.length);
+#                 float[] sum = new float[size];
+#                 for (int i = 0; i < size; i++)
+#                     sum[i] = left[i] + right[i];
+#                 return type.decomposeAsFloat(sum);
+#             }
+
+#             @Override
+#             public Arguments newArguments(ProtocolVersion version)
+#             {
+#                 return new FunctionArguments(version,
+#                                              (v, b) -> type.composeAsFloat(b),
+#                                              (v, b) -> type.composeAsFloat(b));
+#             }
+#         });
+
+#         createTable(KEYSPACE, "CREATE TABLE %s (pk int primary key, value vector<float, 2>)");
+#         execute("INSERT INTO %s (pk, value) VALUES (0, ?)", vector);
+#         execute("INSERT INTO %s (pk, value) VALUES (1, ?)", vector);
+
+#         Object[][] expected = { row(vector(2f, 4f)), row(vector(2f, 4f)) };
+#         assertRows(execute("SELECT f(value, [1.0, 2.0]) FROM %s"), expected);
+#         assertRows(execute("SELECT f([1.0, 2.0], value) FROM %s"), expected);
+#     }
+
+# Test commented out because Scylla doesn't support token(*literal*) syntax.
```

Comment:
This is https://github.com/scylladb/scylladb/issues/12607 or https://github.com/scylladb/scylladb/issues/5411. I think this test can be uncommented and xfail with the reason listed as `#5411` or something.